### ISSUE TITLE
Can't stop codecell kernel

### ISSUE THREAD

**Issue by daejong123** [jupyterlab/jupyterlab#6945]
Hey, I met a problem. maybe is bug.
This error stable repetition in windows system!

the codecell's code is as follows
```
while True:
    print(input('yourname:'))
```

when I stop this codecell,  the error msg is as follows:

```
ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.

ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.

ERROR:root:Invalid alias: The name clear can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name more can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name less can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name man can't be aliased because it is another magic command.
```

**Comment by jasongrout** [comment]
On macOS, with conda packages:

```
# Name                    Version                   Build  Channel
appnope                   0.1.0                 py37_1000    conda-forge
attrs                     19.1.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
bzip2                     1.0.8                h01d97ff_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
ipykernel                 5.1.1            py37h5ca1d4c_0    conda-forge
ipython                   7.7.0            py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.2                      py_0    conda-forge
jedi                      0.14.1                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
json5                     0.8.5                      py_0    conda-forge
jsonschema                3.0.1                    py37_0    conda-forge
jupyter_client            5.3.1                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterlab                1.0.4                    py37_0    conda-forge
jupyterlab_server         1.0.0                      py_1    conda-forge
libcxx                    8.0.0                         4    conda-forge
libcxxabi                 8.0.0                         4    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
markupsafe                1.1.1            py37h1de35cc_0    conda-forge
mistune                   0.8.4           py37h1de35cc_1000    conda-forge
nbconvert                 5.5.0                      py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
notebook                  6.0.0                    py37_0    conda-forge
openssl                   1.1.1c               h01d97ff_0    conda-forge
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.5.1                      py_0    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.2.1                   py37_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pygments                  2.4.2                      py_0    conda-forge
pyrsistent                0.15.3           py37h01d97ff_0    conda-forge
python                    3.7.3                h93065d6_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pyzmq                     18.0.2           py37hee98d25_2    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.29.0               hb7d70f7_0    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                   py_1001    conda-forge
tk                        8.6.9             h2573ce8_1002    conda-forge
tornado                   6.0.3            py37h01d97ff_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
widgetsnbextension        3.4.2                 py37_1000    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zlib                      1.2.11            h01d97ff_1005    conda-forge
```
I evaluate your cell, then hit the stop button in the notebook toolbar. I get an error and one more input. I type in the input and press return, and the kernel is indeed interrupted without any errors like you show.

What version of ipykernel do you have? Can you set up a minimal environment and reproduce the issue with the latest packages?

```
conda create -n jlab6945 -c conda-forge jupyterlab
```

**Comment by daejong123** [comment]
On my macOS, it's ok,
But on windows 10, the errors  stable repetition;
I use pip install jupyter etc.
My windows10 env is as follows:

```
ipykernel          5.1.1
```

```
adafruit-ampy      1.0.7
astroid            2.2.5
attrs              19.1.0
backcall           0.1.0
bleach             3.1.0
certifi            2019.6.16
chardet            3.0.4
Click              7.0
colorama           0.4.1
decorator          4.4.0
defusedxml         0.6.0
entrypoints        0.3
idna               2.8
ipykernel          5.1.1
ipython            7.6.0
ipython-genutils   0.2.0
ipywidgets         7.4.2
isort              4.3.18
jedi               0.14.0
Jinja2             2.10.1
json5              0.8.4
jsonschema         3.0.1
jupyter            1.0.0
jupyter-client     5.2.4
jupyter-console    6.0.0
jupyter-core       4.5.0
jupyterlab         1.0.4
jupyterlab-mfe     0.0.1
jupyterlab-server  1.0.0
lazy-object-proxy  1.3.1
MarkupSafe         1.1.1
mccabe             0.6.1
mistune            0.8.4
nbconvert          5.5.0
nbformat           4.4.0
notebook           5.7.8
pandocfilters      1.4.2
parso              0.5.0
pickleshare        0.7.5
pip                19.0.3
prometheus-client  0.7.1
prompt-toolkit     2.0.9
Pygments           2.4.2
pylint             2.3.1
pyrsistent         0.15.2
pyserial           3.4
python-dateutil    2.8.0
python-dotenv      0.10.3
pywinpty           0.5.5
pyzmq              18.0.2
qtconsole          4.5.1
requests           2.22.0
Send2Trash         1.5.0
setuptools         40.8.0
six                1.12.0
terminado          0.8.2
testpath           0.4.2
tornado            6.0.3
traitlets          4.3.2
typed-ast          1.3.5
urllib3            1.25.3
wcwidth            0.1.7
webencodings       0.5.1
widgetsnbextension 3.4.2
wonderbits         5.5.9
```

```
C:\Windows\system32>ipython -V
7.6.0
```

```
C:\Windows\system32>python3 -V
Python 3.7.3
```

**Comment by jasongrout** [comment]
Can you try with the latest packages? (for example, your ipython is not the latest version)

**Comment by daejong123** [comment]
Update ipython(latest) works on Windows 10.
But I can not stop cell immediately when code contains input, It need again type in the input and press return, then the kernel is indeed interrupted.

**Comment by vidartf** [comment]
The interrupt feature of jupyter on Windows has a few issues. Typically it needs some sort of "kick" after an interrupt (there is also a timer that should regularly provide such a kick, but you might have to wait 5-10 seconds). Often a double interrupt can be useful, or that might just be wishful thinking on my part.

**Comment by vidartf** [comment]
Note: I tested this locally on my Windows machine:
- Running `input()` and then trying to interrupt the kernel (with the "stop" button), I indeed have to finish completing the input. On resumption, the kernel is not interrupted, rather, the cell gets an output with a `KeyboardInterrupt` stack trace.
- The behavior is the same for both lab and classic.
- The server does log `Kernel interrupted` immediately after an interrupt request is sent, so it shouldn't be a front end issue.
